packages/plugins/c-sharp/apollo-dotnet/src/visitor.ts

```diff
@@ -0,0 +1,287 @@
+import { ClientSideBaseVisitor, ClientSideBasePluginConfig, DocumentMode, LoadedFragment, indentMultiline } from '@graphql-codegen/visitor-plugin-common';
+import autoBind from 'auto-bind';
+import { OperationDefinitionNode, print, visit, GraphQLSchema, Kind } from 'graphql';
+import { ApolloDotNetRawPluginConfig } from './config';
+import { camelCase } from 'camel-case';
+import { Types } from '@graphql-codegen/plugin-helpers';
+
+const R_NAME = /name:\s*"([^"]+)"/;
+
+function R_DEF(directive: string) {
+  return new RegExp(`\\s+\\@${directive}\\([^)]+\\)`, 'gm');
+}
+
+export interface ApolloDotNetPluginConfig extends ClientSideBasePluginConfig {
+  ngModule?: string;
+  namedClient?: string;
+  serviceName?: string;
+  serviceProvidedInRoot?: boolean;
+  sdkClass?: boolean;
+  querySuffix?: string;
+  mutationSuffix?: string;
+  subscriptionSuffix?: string;
+}
+
+export class ApolloDotNetVisitor extends ClientSideBaseVisitor<ApolloDotNetRawPluginConfig, ApolloDotNetPluginConfig> {
+  private _operationsToInclude: {
+    node: OperationDefinitionNode;
+    documentVariableName: string;
+    operationType: string;
+    operationResultType: string;
+    operationVariablesTypes: string;
+    serviceName: string;
+  }[] = [];
+
+  constructor(schema: GraphQLSchema, fragments: LoadedFragment[], rawConfig: ApolloDotNetRawPluginConfig, documents?: Types.DocumentFile[]) {
+    super(
+      schema,
+      fragments,
+      rawConfig,
+      {
+        sdkClass: rawConfig.sdkClass,
+        ngModule: rawConfig.ngModule,
```

Comment:
Do we need that?

Reply:
maybe not, I will check this and let you know once I finished.
sorry not sure how early I could have a look on it. maybe during the easter weekend

Reply:
we did not need that, it has been removed.

Reply:
Thank you! Could you align graphql version as well? So we can get it merged.

Reply:
aligned to graphql 15.0.0